src/query/interpret/awesome_memgraph_functions.cpp

```diff
@@ -1174,6 +1177,41 @@ TypedValue Duration(const TypedValue *args, int64_t nargs, const FunctionContext
   MapNumericParameters<Number>(parameter_mappings, args[0].ValueMap());
   return TypedValue(utils::Duration(duration_parameters), ctx.memory);
 }
+
+std::function<TypedValue(const TypedValue *, int64_t, const FunctionContext &)> UserFunction(
+    const mgp_func &func, std::string &fully_qualified_name) {
+  return [func, fully_qualified_name](const TypedValue *args, int64_t nargs, const FunctionContext &ctx) -> TypedValue {
+    const auto &func_cb = func.cb;
```

Comment:
This inserted would check whether the argument to the function is List, and that is not always the case, it could be whatever we want. Also one thing, I have the argument checking in  `procedure::ConstructArguments`

Reply:
You are right, sorry I misunderstood the check that is performed here